### DISCUSSION TITLE
`CMakePresets.json` and Ninja

### DISCUSSION THREAD

**Discussion by bradenmcd** [microsoft/vscode-cmake-tools#2514] [Q&A] [answered]
Is there a commonly deployed solution to the problem of CMake expecting to be run in a shell configured with `vcvarsall.bat` when configuring and building with the Ninja generator?

Visual Studio has some cleverness here that gets engaged with the `architecture.strategy` and `toolset.strategy` cues in `CMakePresets.json`; but these don't seem to do anything for VSCode. Is there some way to tell VSCode to run `vcvarsall.bat` before executing configure/build? Or do I just need to set `environment` in a configuration to ape what `vcvarsall.bat` would do?

From 50,000 feet, the problem I'm trying to solve is this: is it possible to have Visual Studio and VSCode share a configuration preset in `CMakePresets.json`, or are we just not there, yet?

**Comment by bobbrow** [comment, marked as answer]
This is supposed to work and it does for me in the projects I've tried.  We do have code to initialize the MSVC environment when using Ninja with presets (similar to VS) but it sounds like you may have hit a case that didn't work.  If you could open an issue and share your presets file and which versions of Visual Studio you have installed, we can start an investigation from that point.

**Reply by bradenmcd** [reply]
Thank you. I've opened #2516.
